### PR TITLE
Updated section 6.1.6 to include better formating and links to govern…

### PR DESCRIPTION
**Pull request by pgporada**
…ment documents.

Regarding FIPS 186-3, it's been superseded by FIPS 186-4 per http://csrc.nist.gov/publications/drafts/fips186-3/change-notice_fips-186-3.pdf. Does boulder need an update as a result of that?

### CODE REVIEW

File: CPS.md [hunk 1]
```diff
@@ -814,9 +814,18 @@ ISRG CA intermediate Private Keys are RSA keys at least 2048 bits in length.
 
 ### 6.1.6 Public key parameters generation and quality checking
 
-ISRG uses HSMs conforming to FIPS 186-4, capable of providing random number generation and on-board creation of at least 2048-bit RSA keys.
+ISRG uses HSMs conforming to [FIPS 186-4](http://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf), capable of providing random number generation and on-board creation of at least 2048-bit RSA keys.
```
Comment (bdaehlie) [inline CPS.md]: Looks good.

File: CPS.md [hunk 2]
```diff
@@ -814,9 +814,18 @@ ISRG CA intermediate Private Keys are RSA keys at least 2048 bits in length.
...
 
-Per Section 5.3.3, NIST SP 800‐89, the CA ensures that the public exponent of the RSA Keys for a DV-SSL Certificates is in the range between 2<sup>16</sup>+1 and 2<sup>256</sup>-1. The moduli are an odd number, not the power of a prime, and have no factors smaller than 752.
+Per Section 5.3.3, _(Explicit) Partial Public Key Validation for RSA_, of the [NIST SP 800‐89](http://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-89.pdf), the [CA](https://github.com/letsencrypt/boulder) ensures, at a minimum, that
```
Comment (bdaehlie) [inline CPS.md]: Don't link "CA" to boulder.

File: CPS.md [hunk 3]
```diff
@@ -814,9 +814,18 @@ ISRG CA intermediate Private Keys are RSA keys at least 2048 bits in length.
...
+* The length of the modulus is one of the specified values in [FIPS 186-3](http://csrc.nist.gov/publications/drafts/fips186-3/change-notice_fips-186-3.pdf)
+* The value of the public exponent is in the valid range, as specified in [FIPS 186-3](http://csrc.nist.gov/publications/drafts/fips186-3/change-notice_fips-186-3.pdf)
+* The modulus and public exponent are odd numbers
+* The modulus is composite, but not a power of a prime
+* The modulus has no factors smaller than 752.
+
+Additionally,
+
+* The public exponent of the RSA Keys for a DV-SSL Certificates is in the range between 2<sup>16</sup>+1 and 2<sup>256</sup>-1.
```
Comment (bdaehlie) [inline CPS.md]: I find this unnecessarily verbose, both in terms of formatting and content. @jsha should decide on the content at least.

File: CPS.md (hunk 2 above)
Comment (pgporada) [inline CPS.md]: Fixed

File: CPS.md [hunk 4]
```diff
@@ -814,9 +814,18 @@ ISRG CA intermediate Private Keys are RSA keys at least 2048 bits in length.
 
 ### 6.1.6 Public key parameters generation and quality checking
 
-ISRG uses HSMs conforming to FIPS 186-4, capable of providing random number generation and on-board creation of at least 2048-bit RSA keys.
+ISRG uses HSMs conforming to [FIPS 186-4](http://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf), capable of providing random number generation and on-board creation of at least 2048-bit RSA keys.
 
-Per Section 5.3.3, NIST SP 800‐89, the CA ensures that the public exponent of the RSA Keys for a DV-SSL Certificates is in the range between 2<sup>16</sup>+1 and 2<sup>256</sup>-1. The moduli are an odd number, not the power of a prime, and have no factors smaller than 752.
+Per Section 5.3.3, _(Explicit) Partial Public Key Validation for RSA_, of the [NIST SP 800‐89](http://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-89.pdf), the CA ensures, at a minimum, that
+* The length of the modulus is one of the specified values in [FIPS 186-3](http://csrc.nist.gov/publications/drafts/fips186-3/change-notice_fips-186-3.pdf)
```
Comment (jsha) [inline CPS.md]: This links to a change notice rather than the doc itself.

Comment (pgporada) [inline CPS.md]: Regarding FIPS 186-3, it's been superseded by FIPS 186-4 per that change order. Do you still want me to link it to the real FIPS 186-3?

Comment (jsha) [inline CPS.md]: I think if you revert this to match the BRs, the only link will be to NIST SP 800-89.

Comment (pgporada) [inline CPS.md]: Done. Thank you for the clarification. :+1:

**Comment by jsha** [comment]
Sorry, I probably wasn't clear enough - the incorrect link was something I spotted during review, but I think it winds up being irrelevant because we shouldn't accept your diffs to that paragraph. Instead we should say something like the BRs:

> The CA SHALL confirm that the value of the public exponent is an odd number equal to 3 or more. Additionally, the public exponent SHOULD be in the range between 2<sup>16</sup>+1 and 2<sup>256</sup>-1. The modulus SHOULD also have the following characteristics: an odd number, not the power of a prime, and have no factors smaller than 752. [Source: Section 5.3.3, NIST SP 800-89]

This requires a bit of rewriting to match the CPS format and say what we *do* do, not what we SHOULD do, but these are the basic commitments we make to key checking. Also, we don't yet check "power of a prime" (we should, but it's not required). So:

> The CA ensures that the public exponent of any RSA key for a DV-SSL Certificate is in the range between 2<sup>16</sup>+1 and 2<sup>256</sup>-1.  The CA also ensures that the modulus is an odd number and has no factors smaller than 752.

The relevant document to cite, if any, would be the BRs, since they import a subset of NIST SP 800-89. But the BRs are sort of implicitly cited through this whole document (we should make sure to reference them somewhere near the top).

**Comment by pgporada** [comment]
Abandoning this.

**Pull request closed without merging**